godel_simple_gui/src/blending_widget.cpp

```diff
@@ -0,0 +1,120 @@
+#include "godel_simple_gui/blending_widget.h"
+
+// UI specific
+#include "ui_blending_widget.h"
+// General ROS stuff
+#include <ros/console.h>
+#include "godel_msgs/SurfaceBlendingParameters.h"
+// States
+#include "godel_simple_gui/states/scan_teach_state.h"
+
+godel_simple_gui::BlendingWidget::BlendingWidget(QWidget* parent)
+  : QWidget(parent)
+  , active_state_(NULL)
+{
+  // UI setup
+  ui_ = new Ui::BlendingWidget;
+  ui_->setupUi(this);
+
+  options_ = new OptionsSubmenu();
+  options_->hide();
+
+  // Starts in scan teach state
+  changeState( new ScanTeachState() );
+
+  // Wire in buttons
+  connect(ui_->pushButtonNext, SIGNAL(clicked()), this, SLOT(onNextButton()));
+  connect(ui_->pushButtonBack, SIGNAL(clicked()), this, SLOT(onBackButton()));
+  connect(ui_->pushButtonReset, SIGNAL(clicked()), this, SLOT(onResetButton()));
+  connect(ui_->pushButtonOptions, SIGNAL(clicked()), this, SLOT(onOptionsButton()));
+
+  // Wire in Option signals
+  connect(options_, SIGNAL(saveRequested()), this, SLOT(onOptionsSave()));
+
+  // Connect to ROS services
+  loadParameters();
+}
+
+godel_simple_gui::BlendingWidget::~BlendingWidget()
+{
+  delete active_state_;
+  delete options_;
```

Comment:
My understanding is that Qt will delete trees of nodes that are connected to one another via the `parent` argument you see in many cases. Note that `ui_` takes `this` as an argument. This is handled through some kind of signal magic. 

So when you don't give it a parent, you should still manually delete it if you don't want to leak memory.